effects/drugs_effect.py

```diff
@@ -0,0 +1,43 @@
+import message_text_config as msg
+import utils
+from citizens.citizen import Citizen
+from citizens.player import Player
+from game import Game
+
+from effects.effect import Effect
+
+
+class DrugsEffect(Effect):
+    def __init__(self, game: Game, name: str, creator: Citizen) -> None:
+        super().__init__(game, name, creator, 11)
+
+    def _activate_impl(self) -> bool:
+        target_number = utils.read_target_number(
+            self.context, msg.DrugsMessages.ACTIVATION_CHOOSE_TARGET,
+            self._validate)
+        self.targets.append(self.game.citizens[target_number - 1])
+
+        self.user_interaction.save_active(
+            msg.DrugsMessages.ACTIVATION_SUCCESS.format(self.targets[0].name))
+
+        return True
+
+    def _resolve_impl(self) -> bool:
+        for effect in self.targets[0].effects:
+            if effect.__class__.__name__ == "TrapEffect" or \
+                effect.__class__.__name__ == "AlarmEffect" or \
+                    effect.__class__.__name__ == "VideocameraEffect" or \
+                        effect.__class__.__name__ == "WitnessDefenceEffect":
```

Comment:
It's better to use list instead of one-by-one comparison.

Review utils module: is_action_effect method for example.

smth like this:
type(effect).__name__ in ["TrapEffect", "AlarmEffect", "VideocameraEffect", "WitnessDefenceEffect"]

Reply:
Done